R/rglobi.R

```diff
@@ -257,18 +257,16 @@ get_interactions_by_taxa <- function(sourcetaxon, targettaxon = NULL, interactio
   requesturl <- paste(requesturlbase, requestsequence, create_bbox_param(bbox), includeobservations, "type=csv", sep="&")
   result <- read_csv(requesturl)
   
-  if(nrow(result) == 1024){
-    if(ifelse(!"limit" %in% names(otherkeys), TRUE, otherkeys$limit > 1024)){
-      testseq <- requestsequence
-      testseq <- ifelse(grepl("limit=", testseq), gsub("limit=.*?(&|$)", "limit=1\\1", testseq), paste(testseq, "&limit=1", sep=""))
-      testseq <- ifelse(grepl("skip=", testseq), gsub("skip=.*?(&|$)", 
-                                                      paste0("skip=", ifelse("skip" %in% names(otherkeys), otherkeys$skip+1024, 1024), "\\1"),
-                                                      testseq), 
-                        paste(testseq, "&skip=1024", sep=""))
-      testdf <- read_csv(paste(requesturlbase, testseq, create_bbox_param(bbox), includeobservations, "type=csv", sep="&"))
-      if(nrow(testdf) != 0){
-        warning ("Results limit reached. Query generated more results than can be returned via the Web API. Use pagination to retrieve all results.")
-      }
+  if(nrow(result) == 1024 & !"limit" %in% names(otherkeys)){
+    testseq <- paste0(requestsequence, "&limit=1")
+    testseq <- ifelse(grepl("skip=", testseq), 
+                      gsub("skip=.*?(&|$)", 
+                           paste0("skip=", ifelse("skip" %in% names(otherkeys), otherkeys$skip+1024, 1024), "\\1"), 
+                           testseq), 
+                      paste(testseq, "&skip=1024", sep=""))
+    testdf <- read_csv(paste(requesturlbase, testseq, create_bbox_param(bbox), includeobservations, "type=csv", sep="&"))
+    if(nrow(testdf) != 0){
+      warning ("Default results limit reached. Query generated more results than can be returned via the Web API. Increase limit and/or use pagination to retrieve all results.")
```

Comment:
@H-Waters nice! Thanks for taking a stab at this. I was wondering why you make a second query. If a default limit was reached, you'd have to check for the next "page" regardless. Why not just log the warning if default limit has been reached and no limit/skip params were defined.

Reply:
Also, as far as the warning goes, I'd suggest to point to the vignette and mention the skip/limit params. 

Curious to hear your thoughts. How do you suggest moving forward? Eager to incorporate your changes. . .

Reply:
The logic was to cover the scenario in which the total number of results returned happened to be 1024 - with no additional results on subsequent pages. But I'm realising now that this is perhaps unnecessarily cautious/a bit over-engineered! Happy to simplify it to just the check for `nrow(result) == 1024 & !"limit" %in% names(otherkeys)`. 

Also, correct me if you think otherwise, but I think it's only necessary to check whether the user has specified a limit, rather a skip param - since a query with only a skip param specified could still run into the default limit, regardless of where it 'starts'.

Re: the warning text, how about: `"Default results limit reached. Query generated more results than can be returned via the Web API.\nConsider increasing limit parameter and/or using pagination to retrieve all results. See rglobi vignette section on pagination for help modifying limit/skip parameters."` Is this too long? 

Thanks for the feedback!

Reply:
@H-Waters thanks for taking the time to reply!

> Also, correct me if you think otherwise, but I think it's only necessary to check whether the user has specified a limit, rather a skip param - since a query with only a skip param specified could still run into the default limit, regardless of where it 'starts'.

I agree that whenever a limit is set explicitly, or when a skip is used/set, we can assume that the user knows about pagination. And, I'd say that If a user knows about pagination, then the warning is no longer necessary. 

re: warning text - what do you think about dropping the phrase ```Query generated more results than can be returned via the Web API.``` and shorten the warning msg to:

```
"Default results limit reached. \nConsider increasing limit parameter and/or using pagination to retrieve all results. See rglobi vignette section on pagination for help modifying limit/skip parameters."
```

Reply:
PS. Also, please feel free to update to DESCRIPTION to include yourself as an author of the rglobi package as part of the pull request.

Reply:
> `"Default results limit reached. \nConsider increasing limit parameter and/or using pagination to retrieve all results. See rglobi vignette section on pagination for help modifying limit/skip parameters."`

Looks great - I'll change this now!

> PS. Also, please feel free to update to DESCRIPTION to include yourself as an author of the rglobi package as part of the pull request.

Thanks so much for including me, really generous. I'll make that change too.